docs/aws/audit/redshiftmonitoring/rules/not_using_default_port.mdx

```diff
@@ -24,6 +24,231 @@ PCIDSS
+   ```
+
+2. **Modify the Redshift Cluster Port**:
+   If a cluster is using the default port (5439), modify it to use a different port. For example, to change the port to 5432:
+   ```sh
+   aws redshift modify-cluster --cluster-identifier <your-cluster-identifier> --port 5432
+   ```
+
+3. **Verify the Changes**:
+   After modifying the port, verify that the changes have been applied correctly.
+   ```sh
+   aws redshift describe-clusters --cluster-identifier <your-cluster-identifier> --query 'Clusters[*].{ClusterIdentifier:ClusterIdentifier, Port:Port}'
+   ```
+
+4. **Update Security Groups**:
+   Ensure that the security groups associated with the Redshift cluster allow traffic on the new port. Modify the security group rules if necessary.
+   ```sh
+   aws ec2 authorize-security-group-ingress --group-id <your-security-group-id> --protocol tcp --port 5432 --cidr <your-cidr-block>
+   ```
+
+By following these steps, you can prevent Redshift clusters from using the default port using AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent Redshift clusters from using the default port (5439) using Python scripts, you can use the AWS SDK for Python (Boto3). Here are the steps to achieve this:
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed in your Python environment. You can install it using pip if you haven't already:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Initialize a Boto3 session with your AWS credentials and region.
+
+3. **Check Existing Redshift Clusters**:
+   Retrieve the list of existing Redshift clusters and check their port configurations.
+
+4. **Modify Redshift Cluster Port**:
+   If any cluster is using the default port (5439), modify the cluster to use a different port.
+
+Here is a Python script that demonstrates these steps:
+
+```python
+import boto3
+
+# Initialize a Boto3 session
+session = boto3.Session(
+    aws_access_key_id='YOUR_AWS_ACCESS_KEY',
+    aws_secret_access_key='YOUR_AWS_SECRET_KEY',
+    region_name='YOUR_AWS_REGION'
+)
+
+# Create a Redshift client
+redshift_client = session.client('redshift')
+
+# Function to check and modify Redshift cluster port
+def modify_redshift_port(cluster_identifier, new_port):
+    try:
+        # Describe the cluster to get its current configuration
+        response = redshift_client.describe_clusters(ClusterIdentifier=cluster_identifier)
+        cluster = response['Clusters'][0]
+        
+        # Check if the cluster is using the default port
+        if cluster['Endpoint']['Port'] == 5439:
+            print(f"Cluster {cluster_identifier} is using the default port 5439. Modifying to port {new_port}.")
+            
+            # Modify the cluster to use a different port
+            redshift_client.modify_cluster(
+                ClusterIdentifier=cluster_identifier,
+                Port=new_port
+            )
+            print(f"Cluster {cluster_identifier} port modified to {new_port}.")
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>